### PR TITLE
Allow passing empty bodies for "void" headers

### PR DESCRIPTION
**Pull request by Page-**
This restores the behavior of 2.x and 3.0.0

### CODE REVIEW

File: test/pack.js [hunk 1]
```diff
@@ -124,6 +124,29 @@ test('types', function (t) {
...
+  pack.entry({
+    name: 'directory',
+    mtime: new Date(1387580181000),
+    type: 'directory',
+    mode: 0o755,
+    uname: 'maf',
+    gname: 'staff',
+    uid: 501,
+    gid: 20
+  }, "")
```
Comment (mafintosh) [inline test/pack.js:141]: '' not ""

Comment (Page-) [reply test/pack.js]: Fixed

File: test/pack.js [hunk 2]
```diff
@@ -124,6 +124,29 @@ test('types', function (t) {
...
+    gname: 'staff',
+    uid: 501,
+    gid: 20
+  }, '')
+
+  pack.finalize()
+
+  pack.resume()
+
+  t.pass()
```
Comment (mafintosh) [inline test/pack.js:147]: wait for the stream to close before passing

```js
pack.on('error', () => t.fail('should not throw'))
pack.on('close', () => t.pass('closed'))
```

Comment (mafintosh) [reply test/pack.js:147]: otherwise good!

Comment (Page-) [reply test/pack.js]: Done

**Comment by mafintosh** [comment]
Thanks, 3.1.6

**Comment by Page-** [comment]
Thanks!

**Pull request merged**